### PR TITLE
feat: add ability reaply config with migration #2924

### PR DESCRIPTION
**Pull request by BorysTheDev**

### CODE REVIEW

File: src/server/cluster/cluster_family.cc [hunk 1]
```diff
@@ -494,12 +494,35 @@ void ClusterFamily::DflyClusterConfig(CmdArgList args, ConnectionContext* cntx)
...
+
+  SlotRanges enable_slots, disable_slots;
+  {
+    std::lock_guard lk(migration_mu_);
+    // If migration state is changed simultaneously, the changes to config will be applied after
+    // set_config_mu is unlocked and even if we apply the same changes 2 times it's not a problem
+    for (const auto& m : incoming_migrations_jobs_) {
+      if (m->GetState() == MigrationState::C_FINISHED) {
+        const auto& slots = m->GetSlots();
+        enable_slots.insert(enable_slots.end(), slots.begin(), slots.end());
```
Comment (chakaz) [inline src/server/cluster/cluster_family.cc:508]:
```suggestion
        enable_slots = m->GetSlots();
```
Can we do this (also below)? They seem to be of the same type, am I missing something?

File: tests/dragonfly/cluster_test.py [hunk 2]
```diff
@@ -1320,6 +1320,74 @@ async def list_counter(key, client: aioredis.RedisCluster):
     )
 
 
+@dfly_args({"proactor_threads": 4, "cluster_mode": "yes"})
+async def test_cluster_config_reapply(df_local_factory: DflyInstanceFactory):
+    """Check data migration from one node to another."""
```
Comment (chakaz) [inline tests/dragonfly/cluster_test.py:1325]: The entire setup here (~40 lines) is identical to below
maybe we can have a test fixture (is that how it's called?) with setup functions and config apply etc that could reduce the noise here?

Comment (BorysTheDev) [reply tests/dragonfly/cluster_test.py:1325]: I'm not sure what it should look like. I would prefer to have the result as soon as possible and then do tests refactoring

File: src/server/cluster/cluster_family.cc (hunk 1 above)
Comment (chakaz) [reply src/server/cluster/cluster_family.cc:509]: Sorry, my bad, please ignore. I thought we could just assign, but we're inside a `for` loop so we need to append to the vector...

**Review by chakaz — APPROVED** [review]

**Pull request merged**